### PR TITLE
Mk 312 add signatory architecture page

### LINKED ISSUE #312: Create an Architecture Page for Signatory Docs
Add a page to Signatory docs that describe the Signatory Architecture. Use the C4 Model Methodology and include mermaid diagrams. It can include signing flows as a component.

### PR DESCRIPTION
**Pull request by michaelkernaghan**
added  architecture diagrams for Signatory. Includes a sequence diagram for signing flows.

Closes #312

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ The encrypted private key can be obtained from the `.octez-client/` directory
 
 ## Importing the generated key using signatory-cli
 
-A private key can be imported into any of the backend vaults (except: AWS,file & ledger) using the below command.
+A private key can be imported into any of the backend vaults (except: AWS & ledger) using the below command.
```
Comment (AbineshECAD) [inline docs/cli.md:36]: Is the signatory supporting import for the file vault now?

**Review by AbineshECAD — CHANGES_REQUESTED** [review]

File: docs/signatory-architecture.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+Signature service->>+Vault: Request signing operation
+end
+
+Vault->>+Signature service : Send Error or Signature
+Signature service->>+Client: Send Error or Signature
+```
+
+### 4. Tezos Signing Component Model
+
+Tezos uses elliptic curve cryptography to manage private/public key addresses, sign data, and check signatures. Signing a transaction involves hashing the operation request, prefixing it with a watermark, and then signing the resulting byte string with the user's secret key. The signature is then appended to the operation request to create a signed transaction, which can be broadcast to the network for confirmation.
```
Comment (stephengaudet) [inline docs/signatory-architecture.md:205]: the text description says first hash, then prepend watermark, the diagram has the order of those 2 reversed. please make consistent and correct the ordering of the hash and watermark steps

Comment (stephengaudet) [inline docs/signatory-architecture.md:205]: the text description uses the term watermark once, and I see the term "magic byte" in diagrams twice.  please make consistent. if watermark is the correct jargon to use, IMO it sounds better than magic.

**Review by stephengaudet — CHANGES_REQUESTED** [review]

File: docs/cli.md (hunk 1 above)
Comment (michaelkernaghan) [reply docs/cli.md:36]: file vault not included

Comment (AbineshECAD) [reply docs/cli.md:36]: File vault should be there in the `except:` list

**Review by AbineshECAD — CHANGES_REQUESTED** [review]

File: docs/signatory-architecture.md (hunk 2 above)
Comment (stephengaudet) [reply docs/signatory-architecture.md]: I see there is still reference to watermark in the text, and magic byte in the diagrams. I suppose you don't want to change it. 
for my own understanding, are they the same?  what I see in the diagrams for magic byte, is that what is being described as watermark in the text?

**Review by stephengaudet — CHANGES_REQUESTED** [review]

File: docs/signatory-architecture.md (hunk 2 above)
Comment (michaelkernaghan) [reply docs/signatory-architecture.md]: watermark is the old term, which is now used to refer to a counter to guard against double endorsing and baking (high watermark) - now 'magic byte' refers to the prefixed byte for the ops. I'll make it a magic byte throughout t complete the correction.

Comment (michaelkernaghan) [reply docs/signatory-architecture.md]: change was already made but i forgot to update this conversation, pushed already.

File: docs/signatory-architecture.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
+---
+id: architecture
+title: Signatory Architecture
+sidebar_label: Signatory Architecture
+---
+
+# Signatory C4 Architecture Model 
+
+Signatory is the perfect solution for secure and reliable remote signing in Tezos Blockchain cases. With easy-to-use configuration options, users can set up vaults and policies to ensure that only authorized operations are signed. Signatory also supports a variety of hardware-based and cloud-based HSMs, such as [AWS KMS](https://aws.amazon.com/kms/) and [YubiHSM](https://www.yubico.com/?utm_source=google&utm_medium=pd:search&utm_campaign=US_B2C_LeadGen_Google_SEM_Brand&utm_content=&gclid=CjwKCAiAmJGgBhAZEiwA1JZolnAOjpSi_GVY8qz7NyLHASwkXHBu2t1aPNWl2WWHl4Nb2S19OsAWMBoCGucQAvD_BwE), to protect cryptographic keys.
```
Comment (stephengaudet) [inline docs/signatory-architecture.md:9]: "perfect"? the architecture diagram page shouldn't read like a sales and marketing material.

Comment (stephengaudet) [inline docs/signatory-architecture.md:9]: could probably strip the URL parameters from the link to Yubi HSM home page

File: docs/signatory-architecture.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+[Domain-driven design](https://martinfowler.com/tags/domain%20driven%20design.html) ([DDD](https://learn.microsoft.com/en-us/archive/msdn-magazine/2009/february/best-practice-an-introduction-to-domain-driven-design)) is a software development approach emphasizing the importance of understanding the business domain to create effective software solutions. C4 diagrams can help developers understand the domain of the system they are building and design software that aligns with the domain concepts and principles. By using C4 diagrams to visualize the system's architecture, developers can better understand how the system fits into the larger business context and how the different components of the system interact with one another.
+
+The C4 system architectural model is a set of diagrams used to represent the architecture of a software system. There are four levels in the C4 model, each providing an increasing level of detail about the software system.
```
Comment (stephengaudet) [inline docs/signatory-architecture.md:17]: this paragraph is kind of a repeat of 2 paragraphs prior. certainly the "increasing level of detail" is repeated.

File: docs/signatory-architecture.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+
+[C4 architecture diagrams](https://c4model.com/) are visual modelling languages used to communicate software system architectures. They are designed to be simple, easy to understand and provide a high-level view of a system's components and how they interact. C4 diagrams are typically organized into a hierarchy of increasing detail, with each view building on the previous one.
+
+[Domain-driven design](https://martinfowler.com/tags/domain%20driven%20design.html) ([DDD](https://learn.microsoft.com/en-us/archive/msdn-magazine/2009/february/best-practice-an-introduction-to-domain-driven-design)) is a software development approach emphasizing the importance of understanding the business domain to create effective software solutions. C4 diagrams can help developers understand the domain of the system they are building and design software that aligns with the domain concepts and principles. By using C4 diagrams to visualize the system's architecture, developers can better understand how the system fits into the larger business context and how the different components of the system interact with one another.
```
Comment (stephengaudet) [inline docs/signatory-architecture.md:15]: I don't think that architecture diagrams are actually an example of domain diagrams. I think it is a mistake to try to discuss a C4 diagram in the context of DDD.

File: docs/signatory-architecture.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+2. Container Diagram: The container diagram shows the software system as a set of containers, each containing its own set of components. It shows how the different containers interact with each other to form a larger system. This level of the C4 model provides more detail about the software system's components and their interactions.
+3. Component Diagram: The component diagram provides a more detailed view of the software system's components and their interaction. It shows the internal structure of each container and the relationships between the components. This level of the C4 model provides a detailed view of the software system's components and their interactions.
+4. Code Diagram: This is the lowest level of the C4 model, and it provides a detailed view of the code structure of each component. It shows the classes, functions, and other code artifacts that make up each component.
```
Comment (stephengaudet) [inline docs/signatory-architecture.md:22]: Do we need to describe what C4 diagrams are in points 1 through 4 on the Signatory website?  maybe a link to the C4 website is best

File: docs/signatory-architecture.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+
+There is no requirement to detail every container and component. Often the high-level description is enough to convey the required understanding. It is also unnecessary to detail the code level, as a sufficiently detailed component diagram can suffice.
+
+### 1. Signatory System Context
+
+
+For Signatory, a remote signer application for the Tezos blockchain, the C4 system context diagram shows the Signatory software system as the central box. The diagram includes the Signatory user, responsible for setting up the system's configuration, and the client software system, which submits requests to the Tezos API. The diagram also shows various vaults, such as AWS KMS and YubiHSM, which Signatory uses to store cryptographic keys, and the Prometheus service, which stores and tracks metrics data.
```
Comment (stephengaudet) [inline docs/signatory-architecture.md:29]: "pulls and aggregates" may be better than "stores and tracks"

File: docs/signatory-architecture.md [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+Signature service->>+Vault: Request signing operation
+end
+
+Vault->>+Signature service : Send Error or Signature
+Signature service->>+Client: Send Error or Signature
+```
+
+### 4. Tezos Signing Component Model
+
+Tezos uses elliptic curve cryptography to manage private/public key addresses, sign data, and check signatures. Signing a transaction involves prefixing it with a magic-byte, hashing the operation request, and then signing the resulting byte string with the user's secret key. The signature is then appended to the operation request to create a signed transaction, which can be broadcast to the network for confirmation.
```
Comment (stephengaudet) [inline docs/signatory-architecture.md:205]: "key pairs" may be better than "key addresses"

**Review by stephengaudet — CHANGES_REQUESTED** [review]

File: docs/cli.md (hunk 1 above)
Comment (AbineshECAD) [reply docs/cli.md:36]: File vault should be included. please change this.

**Review by AbineshECAD — CHANGES_REQUESTED** [review]

File: docs/signatory-architecture.md (hunk 4 above)
Comment (michaelkernaghan) [reply docs/signatory-architecture.md:17]: removed the repetitive clause ", each providing an increasing level of detail about the software system."

File: docs/signatory-architecture.md (hunk 5 above)
Comment (michaelkernaghan) [reply docs/signatory-architecture.md:15]: The Book on how to do Mermais Diagrams is a book about DDD, and it is where the instruction on how to do a C4 model is.

File: docs/signatory-architecture.md (hunk 6 above)
Comment (michaelkernaghan) [reply docs/signatory-architecture.md:22]: yes, I think it shows clearly what we are discussing and what we want users to understand and provides context they can check back on easily.

File: docs/signatory-architecture.md (hunk 8 above)
Comment (michaelkernaghan) [reply docs/signatory-architecture.md:205]: changed to 'key pairs'

File: docs/signatory-architecture.md (hunk 7 above)
Comment (michaelkernaghan) [reply docs/signatory-architecture.md:29]: changed as requested

File: docs/signatory-architecture.md (hunk 3 above)
Comment (michaelkernaghan) [reply docs/signatory-architecture.md]: removed as requested

File: website/sidebars.js [hunk 9]
```diff
@@ -22,7 +22,7 @@ const sidebars = {
 			className: 'sidebarHeader',
 			collapsed: false,
 			collapsible: false,
-			items: ['start', 'file_based', 'yubihsm', 'azure_kms', 'gcp_kms', 'aws_kms', 'ledger', `cli`, 'remote_policy', 'bakers'],
+			items: ['start', 'file_based', 'yubihsm', 'azure_kms', 'gcp_kms', 'aws_kms', 'ledger', `cli`, 'remote_policy', 'architecture', 'bakers'],
```
Comment (AbineshECAD) [inline website/sidebars.js:25]: Do we need this doc for all users? This document link can be given in getting started, and mention that it is optional for anyone who wants a more profound understanding of signatory.

**Comment by michaelkernaghan** [comment]
yes we need for all users who include reviewers

File: website/sidebars.js (hunk 9 above)
Comment (michaelkernaghan) [reply website/sidebars.js:25]: yes we need for all users who include reviewers

**Review by AbineshECAD — APPROVED** [review]

**Review by stephengaudet — APPROVED** [review]

**Pull request merged**